File: 2023-2024/Gara 2/farmula1.py
points = [0, 25, 18, 15, 12, 10, 8, 6, 4, 2, 1]
input_file = open("input.txt", mode = "r")

t = int(input_file.readline().strip())

for _ in range(t):
    # Leggiamo la riga bianca facendo un readline inutile
    input_file.readline()

    n = int(int(input_file.readline()))
    
    daniel_score = 0
    max_score = 0
    # Lista con le posizioni di Daniel
    results = list(map(int, input_file.readline().split()))
    
    # L'idea è calcolare quanti punti ha fatto Daniel e
    # quanti punti ha fatto il primo in classifica
    # Se Daniel è arrivato primo allora prendiamo il secondo classificato
    for i in range(n):
        position = int(results[i])

        # Se Daniel non è arrivato primo
        if position != 1:
            max_score += 25
        else:
            max_score += 18
 
        if position <= 10:
            daniel_score += points[position]
    
    if daniel_score >= max_score:
        print("Champion")
    else:
        print("Practice harder")

File: 2023-2024/Gara 2/farmula1.cpp
#include <iostream>
using namespace std;

int points[] = {0, 25, 18, 15, 12, 10, 8, 6, 4, 2, 1};

int main()
{
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	
	int t;
	cin >> t;
	
	for(; t; t--)
	{
		int n;
		cin >> n;
		
		int score = 0;
		int max_score = 0;
		
		for(int i = 1; i <= n; i++)
		{
			int nr;
			cin >> nr;
			if(nr != 1)
				max_score += 25;
			else
				max_score += 18;
			if(nr <= 10)
				score += points[nr];
		}
		
		cout << (score >= max_score ? "Champion" : "Practice harder") << '\n';
	}
}

